lib/benchmark/benchmark.go

```diff
@@ -268,6 +334,131 @@ func execute(m benchMeasure) error {
 	return nil
 }
 
+func (c *Config) kubernetesBenchmarkCreator(ctx context.Context, tc *client.TeleportClient) (workloadFunc, error) {
+	tlsClientConfig, err := getKubeTLSClientConfig(ctx, tc)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	restConfig := &rest.Config{
+		Host:            tc.KubeClusterAddr(),
+		TLSClientConfig: tlsClientConfig,
+		APIPath:         "/api",
+		ContentConfig: rest.ContentConfig{
+			GroupVersion:         &schema.GroupVersion{Version: "v1"},
+			NegotiatedSerializer: scheme.Codecs,
+		},
+	}
+
+	// if the user has specified a pod exec benchmark, use that instead of the
+	// default kubernetes client.
+	if c.PodExecBenchmark != nil {
+		exec, err := c.kubeExecOnPod(ctx, tc, restConfig)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return func(bm benchMeasure) error {
+			err := exec.StreamWithContext(ctx, remotecommand.StreamOptions{
+				Stdin:  tc.Stdin,
+				Stdout: tc.Stdout,
+				Stderr: tc.Stderr,
+				Tty:    c.Interactive,
+			})
+			return trace.Wrap(err)
+		}, nil
+	}
+
+	// create a kubernetes client that will be used to execute the benchmark.
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return func(bm benchMeasure) error {
+		// List all pods in all namespaces.
+		_, err := kubeClient.CoreV1().Pods(c.PodNamespace).List(ctx, metav1.ListOptions{})
+		return trace.Wrap(err)
+	}, nil
+}
+
+func (c *Config) kubeExecOnPod(ctx context.Context, tc *client.TeleportClient, restConfig *rest.Config) (remotecommand.Executor, error) {
```

Comment:
I thought about it initially but `tsh kube exec` does a ton of stuff including handling weird cases for supporting TTY mode in windows.
It also supports terminal resizes that we don't want to support here.
Let me know what you think

Reply:
@rosstimothy 

I have mixed feelings about this.

`tsh kube exec` is not a very good measure for `kubectl exec` calls since it does a lot of stuff behind the scenes. Calling `tsh kubectl` is a better indication but since it calls `os.Exec` to re-exec the `tsh kubectl` code it also adds some slowness. 

Calling `kubectl`/`tsh kubectl` also has performance implications because we must read the kubeconfig and the credentials by calling `tsh kube credentials`. The last call must read 3-4 files if credentials are cached which might affect the results if we have many parallel executioners.

I am reusing a connection pool instead of starting a new transport per request. It's not an accurate measure of the real latency we will see in the wild but we can measure the request time more accurately.

Lastly, 99.9% of our users interact via `kubectl` instead of `tsh kube exec`/`tsh kubectl`

Let me know what you think.